hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/RDBTable.java

```diff
@@ -218,14 +219,63 @@ public TableIterator<byte[], KeyValue<byte[], byte[]>> iterator()
   @Override
   public TableIterator<byte[], KeyValue<byte[], byte[]>> iterator(byte[] prefix)
       throws IOException {
-    return new RDBStoreByteArrayIterator(db.newIterator(family, false), this,
-        prefix);
+    RDBStoreByteArrayIterator itr  = new RDBStoreByteArrayIterator(db.newIterator(family, false), this, prefix);
+    return new TableIterator<byte[], KeyValue<byte[], byte[]>>() {
+      @Override
+      public void seekToFirst() {
+        itr.seekToFirst();
+      }
+
+      @Override
+      public void seekToLast() {
+        itr.seekToLast();
+      }
+
+      @Override
+      public KeyValue<byte[], byte[]> seek(byte[] bytes) {
+        try (UncheckedAutoCloseableSupplier<RawKeyValue<byte[]>> kv = itr.seek(bytes)) {
+          return kv == null ? null : kv.get();
+        }
+      }
+
+      @Override
+      public void removeFromDB() throws IOException {
+        itr.removeFromDB();
+      }
+
+      @Override
+      public void close() {
+        itr.close();
+      }
+
+      @Override
+      public boolean hasNext() {
+        return itr.hasNext();
+      }
+
+      @Override
+      public KeyValue<byte[], byte[]> next() {
+        try (UncheckedAutoCloseableSupplier<RawKeyValue<byte[]>> kv = itr.next()) {
+          return kv.get();
+        }
+      }
+    };
   }
 
-  TableIterator<CodecBuffer, KeyValue<CodecBuffer, CodecBuffer>> iterator(
+  public TableIterator<byte[], UncheckedAutoCloseableSupplier<RawKeyValue<byte[]>>> closeableSupplierIterator(
+      byte[] prefix) throws IOException {
+    return new RDBStoreByteArrayIterator(db.newIterator(family, false), this, prefix);
+  }
+
+  TableIterator<CodecBuffer, UncheckedAutoCloseableSupplier<RawKeyValue<CodecBuffer>>> iterator(
       CodecBuffer prefix) throws IOException {
+    return iterator(prefix, 2);
+  }
+
+  TableIterator<CodecBuffer, UncheckedAutoCloseableSupplier<RawKeyValue<CodecBuffer>>> iterator(
```

Comment:
We can remove this iterator creation here and be part of specialized method, as this needs to be wrapped for handling Buffer release. No need expose.